### PR TITLE
Home link block: Add 'current-menu-item'

### LINKED ISSUE #34770: Home link block doesn't use `current-menu-item` class when on front page
## What problem does this address?

When you create a navigation using the `core/navigation` block you often want to add a link to `Home`. It's great that we supply a "Home link" block for this purpose. 

However, whilst normal Nav Link blocks output a `current-menu-item` class when you are on that page, the Home link block does not. This makes it impossible to style the current page consistently across the navigation.

Screen Shot 2021-09-13 at 09 22 45

Screen Shot 2021-09-13 at 09 22 03

## What is your proposed solution?

The home link block should output the `current-menu-item` class when you are on that page. It should mirror the same functionality as the `core/navigation-link` block.

### PR DESCRIPTION
**Pull request by carolinan**
## What?

Adds the CSS class 'current-menu-item' to the home link block.
Testing instructions were updated June 26.

Closes https://github.com/WordPress/gutenberg/issues/34770

## Why?

Adding the class makes the block easier to style. The class is already used in the page list block and the navigation link block, and this change makes it easier to style the current menu items consistently.

## How?

Adds the CSS class, conditionally, to the `<li>` item in the block's index.php file.

## Testing Instructions
Add a navigation block to a site-wide header template part. The important thing is that the menu item can be tested on the correct page to trigger the condition and add the CSS class.

Inside the navigation block, insert the Home link block.
View the homepage on the front of the website.
Confirm that the Home link block has the CSS class `current-menu-item` on the li element:
`<li class="wp-block-navigation-item current-menu-item wp-block-home-link">`
Open any link on the site that is not the homepage, and confirm that the Home menu item no longer has the `current-menu-item` class.

There are two edge cases, and that is when the user adds the assigned "posts page" and or the homepage from the Reading settings to the navigation block, besides the home link block. In this case, the class name as well as the aria-current attribute, will be duplicated.
I think this needs to be considered a user error, adding the same link more than once.

### CODE REVIEW

File: packages/block-library/src/home-link/index.php [hunk 1]
```diff
@@ -94,11 +94,13 @@ function block_core_home_link_build_css_font_sizes( $context ) {
 function block_core_home_link_build_li_wrapper_attributes( $context ) {
 	$colors          = block_core_home_link_build_css_colors( $context );
 	$font_sizes      = block_core_home_link_build_css_font_sizes( $context );
+	$is_active       = is_home() || ( is_front_page() && 'page' === get_option( 'show_on_front' ) );
```
Comment (alexstine) [inline packages/block-library/src/home-link/index.php]: Why not use `is_front_page()` here alone?

https://developer.wordpress.org/themes/basics/conditional-tags/#the-main-page

Looks like it should return true in all home cases? In all my years of dev with WordPress, this is still confusing to me.

**Review by alexstine — COMMENTED** [review]
@carolinan One question below. Also, checks are in a weird state. Needs refresh?

File: packages/block-library/src/home-link/index.php (hunk 1 above)
Comment (carolinan) [reply packages/block-library/src/home-link/index.php]: I used the same condition that was used to add the `aria-current`.

But looking closer, there does seem to be a bug here.
If the reading setting uses a "posts page", and that page is used in the navigation menu, then both the home link block and that page link have `aria-current="page"` and the new class name.
I am going to check if there are any previous discussions around that.

Comment (carolinan) [reply packages/block-library/src/home-link/index.php]: Update:  OK I did not find anything other than that [I was responsible](https://github.com/WordPress/gutenberg/pull/35880) 😬

Comment (carolinan) [reply packages/block-library/src/home-link/index.php]: I made two small changes to the condition, but I think the duplicate `aria-current` and class names need to be thought of as user errors.
The duplication is not wrong: The user should not add the same link more than once to the same navigation block.

**Review by aristath — APPROVED** [review]
I don't know how we missed this when initially adding the `current-menu-item` classes...
Good catch @carolinan, thank you for working on this!

Approving this one, it's good to merge 👍

**Comment by getdave** [comment]
Thank you both. This is so great to see! 👏

**Pull request merged**